### PR TITLE
QMAPS-781 - Geocoder feedback 

### PR DESCRIPTION
**Pull request by Sense**

### CODE REVIEW

File: src/modals/feedback_modal.js [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+      result_other_details: '',
+    }
+
+    this.selected_result = {}
+
+    listen('open_feedback_modal', (results) => {
+      this.current_query = window.__searchInput.searchInputHandle.value
+      this.original_results = this.displayed_results = results || []
+      this.displayed_results.forEach((res, idx) => {
+        res.icon = IconManager.get({className : res.className, subClassName : res.subClassName , type : res.type})
```
Comment (GuillaumeGomez) [inline src/modals/feedback_modal.js]: Please remove whitespace before ':' and before ','.

Comment (GuillaumeGomez) [inline src/modals/feedback_modal.js]: Shouldn't there be a backup in case the icon cannot be found?

File: src/modals/feedback_modal.js [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+    this.ExtendedString = ExtendedString
+
+    this.original_results = []
+    this.displayed_results = []
+    this.current_query = ''
+
+    this.fields = {
+      result_not_found: false,
+      result_not_found_details: '',
```
Comment (GuillaumeGomez) [inline src/modals/feedback_modal.js]: Is it expected to not have `_details` fields for all fields?

File: src/modals/feedback_modal.js [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+      this.current_query = window.__searchInput.searchInputHandle.value
+      this.original_results = this.displayed_results = results || []
+      this.displayed_results.forEach((res, idx) => {
+        res.icon = IconManager.get({className : res.className, subClassName : res.subClassName , type : res.type})
+      })
+      this.modal.open()
+    })
+  }
+
+  toggleInputField (key) {
```
Comment (GuillaumeGomez) [inline src/modals/feedback_modal.js]: Please remove whitespace before '('.

File: src/modals/feedback_modal.js [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+      this.displayed_results = this.displayed_results.filter(res => res.id === result.id)
+    }
+    this.update()
+  }
+
+  update () {
+    this.modal.panel.update()
+  }
+
+  close () {
```
Comment (GuillaumeGomez) [inline src/modals/feedback_modal.js]: Please remove whitespace before '('.

File: src/modals/feedback_modal.js [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+      this.selected_result = {}
+      this.displayed_results = this.original_results
+    } else {
+      this.selected_result = result
+      this.displayed_results = this.displayed_results.filter(res => res.id === result.id)
+    }
+    this.update()
+  }
+
+  update () {
```
Comment (GuillaumeGomez) [inline src/modals/feedback_modal.js]: Please remove whitespace before '('.

File: src/adapters/suggest.js [hunk 6]
```diff
@@ -90,6 +91,7 @@ export default class Suggest {
         if (favorites.length > 0) {
           suggestDom += this.favoritesRender(favorites)
         }
+        if (this.config.feedback.enabled && pois.length > 0) suggestDom += this.feedbackRender()
```
[the remark this answers is not in the document]
Comment (Sense) [reply src/adapters/suggest.js]: Done.
Is this possible to get some codestyle guideline somewhere ? thanks 🤗

File: src/modals/feedback_modal.js (hunk 1 above)
Comment (Sense) [reply src/modals/feedback_modal.js]: Done.

Comment (Sense) [reply src/modals/feedback_modal.js]: Got this from the `suggest.js` as it's already implemented.

File: src/modals/feedback_modal.js (hunk 2 above)
Comment (Sense) [reply src/modals/feedback_modal.js]: Please check the mockup, details are only asked for a certain expected result which has not been found OR from an other chosen reason based on the next checkboxes. 
Capture d’écran 2019-06-03 à 10 09 30

File: src/modals/feedback_modal.js (hunk 3 above)
Comment (Sense) [reply src/modals/feedback_modal.js]: Done.

File: src/modals/feedback_modal.js (hunk 4 above)
Comment (Sense) [reply src/modals/feedback_modal.js]: Done.

File: src/modals/feedback_modal.js (hunk 5 above)
Comment (Sense) [reply src/modals/feedback_modal.js]: Done.

**Comment by Sense** [comment]
> Good as a first poc 👍
> We leave this PR open until we have more time to set up the back systems.

Thanks, glad to hear that and looking forward to see this merged 😎.

**Comment by bbecquet** [comment]
Project priorities have changed and if we still decide to gather user feedback on the geocoder in the future, it will have to be re-implemented in React instead of doTjs which isn't used anymore in the project.
Sorry @Sense, we're closing this :|

**Pull request closed without merging**